python/tidy/servo_tidy/tidy.py

```diff
@@ -632,6 +634,57 @@ def check_webidl_spec(file_name, contents):
     yield (0, "No specification link found.")
 
 
+def duplicate_key_yaml_constructor(loader, node, deep=False):
+    mapping = {}
+    for key_node, value_node in node.value:
+        key = loader.construct_object(key_node, deep=deep)
+        if key in mapping:
+            raise KeyError(key)
+        value = loader.construct_object(value_node, deep=deep)
+        mapping[key] = value
+    return loader.construct_mapping(node, deep)
+
+
+class SafeYamlLoader(yaml.SafeLoader):
```

Comment:
I don't see the benefit of this over using `yaml.SafeLoader` in the qualified form.